### PR TITLE
Fix compilation of cuDF benchmarks with build.sh

### LINKED ISSUE #4820: [BUG] Build libcudf benchmarks in CI
**Describe the bug**
gpuCI does not build libcudf benchmarks, so PRs frequently break them.

**Steps/Code to reproduce bug**
Configure cmake with -DBUILD_BENCHMARKS=ON and build libcudf. This does not happen in CI.

**Expected behavior**
Configure the libcudf build in CI with BUILD_BENCHMARKS=ON so we can catch compile errors in CI.

### PR DESCRIPTION
**Pull request by vuule**
Fixes #4820 
- Added a build target for benchmarks.
- Modified build.sh to build benchmarks when the command line argument is passed.
- Removed some unused CMake code.
- Modified a few lambdas in benchmarks to work around a version-specific gcc issue.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
rerun tests

File: cpp/benchmarks/CMakeLists.txt [hunk 1]
```diff
@@ -13,10 +15,9 @@ function(ConfigureBench CMAKE_BENCH_NAME CMAKE_BENCH_SRC)
     target_link_libraries(${CMAKE_BENCH_NAME} benchmark benchmark_main pthread cudf )
     set_target_properties(${CMAKE_BENCH_NAME} PROPERTIES
                             RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/gbenchmarks")
+    set(BENCHMARK_LIST ${BENCHMARK_LIST} ${CMAKE_BENCH_NAME} CACHE INTERNAL "BENCHMARK_LIST")
 endfunction(ConfigureBench)
 
-option(CMAKE_ENABLE_BENCHMARKS "Enable building cuDF benchmarks" OFF)
```
Comment (harrism) [inline cpp/benchmarks/CMakeLists.txt:18]: So is there no way to disable building them (e.g. when building locally) now?

Comment (vuule) [reply cpp/benchmarks/CMakeLists.txt:18]: This variable was not used in this CMake file at all. I removed what seemed like dead code (see the change in the CMake file in /tests too). 
You can specify whether to build benchmarks in CMake using -DBUILD_BENCHMARKS flag now.

**Comment by vuule** [comment]
rerun tests

**Pull request merged**